modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/delivery/DisruptorBasedContentDecompressionHandler.java

```diff
@@ -0,0 +1,127 @@
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.wso2.andes.kernel.disruptor.delivery;
+
+import com.lmax.disruptor.EventHandler;
+import org.apache.log4j.Logger;
+import org.wso2.andes.kernel.*;
+import org.wso2.andes.kernel.disruptor.ContentCompressionDecompressionHandler;
+
+import java.util.*;
+
+/**
+ * Disruptor handler used to decompress message contents, before sending messages to subscribers.
+ */
+public class DisruptorBasedContentDecompressionHandler implements EventHandler<DeliveryEventData> {
+
+    /**
+     * Class Logger for logging information, error and warning.
+     */
+    private static final Logger log = Logger.getLogger(DisruptorBasedContentDecompressionHandler.class);
+
+    /**
+     * Maximum content chunk size stored in DB
+     */
+    private final int maxChunkSize;
+
+    /**
+     * Creates a {@link org.wso2.andes.kernel.disruptor.delivery.DisruptorBasedContentDecompressionHandler} object
+     *
+     * @param maxContentChunkSize maximum content chunk size stored in DB
+     */
+    public DisruptorBasedContentDecompressionHandler(int maxContentChunkSize) {
+        this.maxChunkSize = maxContentChunkSize;
+    }
+
+    @Override
+    public void onEvent(DeliveryEventData deliveryEventData, long l, boolean b) throws Exception {
+
+        // If the content is already decompressed, there is no point in further decompressing
+        // Therefore ignore
+        if (!deliveryEventData.availableForContentDecompression()) {
+            return;
+        }
+
+        DisruptorCachedContent content = (DisruptorCachedContent) deliveryEventData.getAndesContent();
+        Map<Integer, AndesMessagePart> contentList;
+        List<AndesMessagePart> messagePartsList;
+        Map<Integer, AndesMessagePart> messagePartMap;
+
+        ProtocolMessage metadata = deliveryEventData.getMetadata();
+        int contentSize = metadata.getMessage().getMessageContentLength();
+        int originalMessageSize = contentSize;
+        long messageID = metadata.getMessageID();
+
+        if (content != null) {
+            if (originalMessageSize > ContentCompressionDecompressionHandler.contentCompressionThreshold) {
+
+                    /* If the current message was compressed by the server (if compression is enabled and content size
+                     * of the message is larger than the content compression threshold), decompress
+                     * the message content and, make message parts and put message parts into the map
+                     */
+
+                contentList = content.getContentList();
+                messagePartsList = new ArrayList<AndesMessagePart>();
```

Comment:
To create a DisruptorCachedContent (to set the andesContent of deliveryEventData before sending messages to the subscriber), we need a message part list map. Thus, need to create the map.
I'll write another part to get the compressed message from a map. Then we can use it directly without creating an array list.